### ISSUE TITLE
graphapi cleanFilter failure leads to lsc stop

### ISSUE THREAD

**Issue by davidcoutadeur** [lsc-project/lsc-microsoft-graph-api-plugin#1]
Graph API is very poor for filters:
https://docs.microsoft.com/fr-fr/graph/query-parameters#filter-parameter

```
Note: The following $filter operators are not supported for Azure AD resources: ne, gt, ge, lt, le, and not. The contains string operator is currently not supported on any Microsoft Graph resources.
```

This is very uncomfortable.

For example, I can't exclude these type of entries from the synchronization:

```
Feb 24 15:55:50 - ERROR - Error while synchronizing ID {mail=A.N'DOME@domain.com}: org.lsc.exception.LscServiceException: {
  "error": {
    "code": "BadRequest",
    "message": "Invalid filter clause",
    "innerError": {
      "request-id": "0e8f5835-a894-48e5-99bc-2327bf5205be",
      "date": "2020-02-24T14:55:50"
    }
  }
}
```
The problem is certainly the quote, but it would be nice to have the LSC going on after such failures.

**Comment by rouazana** [comment]
Hi,

I don't think to try to recover on such a failure is a good idea. If there is a configuration issue (here because Graph API does not support some filters), it seems to be a good idea to crash waiting for the user to solve the configuration issue, no?

**Comment by davidcoutadeur** [comment]
Very important remark: this bug appears only for the clean task!

This makes me think to another improvement: could we have a customized clean filter in lsc.xml?

**Comment by davidcoutadeur** [comment]
@rouazana @remk if you can give a look

**Comment by davidcoutadeur** [comment]
I rename the issue because it affects the cleanFilter and not the getOneFilter.

**Comment by rouazana** [comment]
fixed in v1.1
